activerecord/lib/active_record/fixtures.rb

```diff
@@ -840,14 +846,19 @@ def [](key)
 
     alias :to_hash :fixture
 
-    def find
+    def find(config = nil)
+      original_config = ActiveRecord::Base.connection_config.stringify_keys
+      config ||= original_config
+      ActiveRecord::Base.establish_connection(config) unless original_config == config
       if model_class
         model_class.unscoped do
           model_class.find(fixture[model_class.primary_key])
         end
       else
         raise FixtureClassNotFound, "No class attached to find."
       end
+    ensure
+      ActiveRecord::Base.establish_connection(original_config) unless original_config == config
```

Comment:
Do we need to do the `unless` here? Will it explode or be slow if we always est a connection to the original config? Same question for line 852.

Reply:
I don't think its noticeably slower, but many tests break that use `self.use_transactional_tests`, which is true by default. For instance, most of the persistence tests (see `bin/test -n PersistenceTest#test_increment_attribute`). The reload logic still works but when the fixture is reloaded in `#find`, the original record is returned. I'm sure there's a better way of dealing with this, but I haven't found a better solution than this band-aid for now.